distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/UrlResourceReaderConfigurator.java

```diff
@@ -30,38 +34,94 @@
 
   private static final String PID = "ddf.catalog.resource.impl.URLResourceReader";
 
+  private static final String ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY = "rootResourceDirectories";
+
   private ConfigurationAdmin configAdmin;
 
   public UrlResourceReaderConfigurator(ConfigurationAdmin configAdmin) {
     this.configAdmin = configAdmin;
   }
 
   public void setUrlResourceReaderRootDirs(String... rootResourceDirs) throws IOException {
-    Configuration configuration = configAdmin.getConfiguration(PID, null);
-    Dictionary<String, Object> properties = new DictionaryMap<>();
-    Set<String> rootResourceDirectories =
+    Collection<String> newRootResourceDirs =
         ImmutableSet.<String>builder().add(rootResourceDirs).build();
-    properties.put("rootResourceDirectories", rootResourceDirectories);
-    configuration.update(properties);
-    for (int i = 0; i < 5; i++) {
-      Configuration updatedConfig = configAdmin.getConfiguration(PID, null);
-      if (updatedConfig
-          .getProperties()
-          .get("rootResourceDirectories")
-          .equals(rootResourceDirectories)) {
-        break;
-      }
-      boolean interrupted = false;
-      try {
-        TimeUnit.SECONDS.sleep(5);
-      } catch (InterruptedException e) {
-        interrupted = true;
-      } finally {
-        if (interrupted) {
-          Thread.currentThread().interrupt();
-        }
+    Configuration configuration = configAdmin.getConfiguration(PID, null);
+    Optional<Collection<String>> currentRootResourceDirs =
+        getCurrentRootResourceDirs(configuration);
+
+    if (currentRootResourceDirs.isPresent()) {
+
+      LOGGER.debug(
+          "{} {}, current value: {}, new value: {}",
+          PID,
+          ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY,
+          currentRootResourceDirs.get(),
+          newRootResourceDirs);
+
+      if (rootResourceDirsEqual(currentRootResourceDirs.get(), newRootResourceDirs)) {
+        LOGGER.debug(
+            "{} {} unchanged, skipping update", PID, ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY);
+        return;
       }
     }
-    LOGGER.info("URLResourceReader props after update: {}", configuration.getProperties());
+
+    updateUrlResourceReaderRootDirs(configuration, newRootResourceDirs);
+  }
+
+  @SuppressWarnings("unchecked")
+  private Optional<Collection<String>> getCurrentRootResourceDirs(Configuration configuration) {
+    Dictionary<String, Object> properties = configuration.getProperties();
+
+    if (properties == null) {
+      return Optional.empty();
+    }
+
+    Object currentRootResourceDirs = properties.get(ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY);
+
+    if (!(currentRootResourceDirs instanceof Collection)) {
+      return Optional.empty();
+    }
+
+    return Optional.of((Collection) currentRootResourceDirs);
```

Comment:
✏️ To avoid the suppression: `return Optional.of((Collection<String>)currentRootResourceDirs);`

Reply:
Because of runtime type erasure, you get a warning even if you add the type to the `Collection`.